# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: tools/profiler/public/ProfilerControl.h
# Commit: 1f4dc2487e0e
# Full Hash: 1f4dc2487e0e822691d70c2b5337a6b664325752
# Author: Adam Brouwers-Harries <abrouwersharries@mozilla.com>
# Date: 2024-04-03 09:34:09
# Regressor Bug: 1865930
# File Overlap Count: 1
# Description:
#   Bug 1865930: Add support for stopping the profiler using UNIX signals r=canaltinova,profiler-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D203146
# ==============================================================================

diff -r 58a369871836 -r 1f4dc2487e0e tools/profiler/public/ProfilerControl.h
--- a/tools/profiler/public/ProfilerControl.h	Tue Apr 02 18:20:20 2024 +0000
+++ b/tools/profiler/public/ProfilerControl.h	Tue Apr 02 18:35:50 2024 +0000
@@ -123,6 +123,18 @@
     const char** aFilters, uint32_t aFilterCount, uint64_t aActiveTabID,
     const mozilla::Maybe<double>& aDuration = mozilla::Nothing());
 
+// Tell the profiler to look up the download directory for writing profiles.
+// With some features, such as signal control, we need to know the location of
+// a directory where we can save profiles to disk. Because we start the
+// profiler before we start the directory service, we can't access the
+// download directory at profiler startup. Similarly, when we need to get the
+// directory, we often can't, as we're running in non-main-thread contexts
+// that don't have access to the directory service. This function gives us a
+// third option, by giving us a hook to look for the download directory when
+// the time is right. This might be triggered internally (e.g. when we start
+// profiling), or externally, e.g. after the directory service is initialised.
+void profiler_lookup_download_directory();
+
 //---------------------------------------------------------------------------
 // Control the profiler
 //---------------------------------------------------------------------------